tket/tests/Circuit/test_Circ.cpp

```diff
@@ -1703,6 +1735,35 @@ SCENARIO("Decomposing a multi-qubit operation into CXs") {
             0, 0, sq, sq;
     // clang-format on
     REQUIRE((u - correct).cwiseAbs().sum() < ERR_EPS);
+    REQUIRE(rep.count_gates(OpType::CX) == 2);
+  }
+  GIVEN("A CRy(+-pi) gate") {
+    Circuit circ(2);
+    Vertex v;
+    WHEN("CRy(+pi)") { v = circ.add_op<unsigned>(OpType::CRy, 1., {0, 1}); }
+    WHEN("CRy(-pi)") { v = circ.add_op<unsigned>(OpType::CRy, -1., {0, 1}); }
+    const Op_ptr op = circ.get_Op_ptr_from_Vertex(v);
+    Circuit rep;
+    rep = CX_circ_from_multiq(op);
+
+    REQUIRE(rep.count_gates(OpType::CX) == 1);
+
+    const auto u = tket_sim::get_unitary(rep);
+    const auto u_correct = tket_sim::get_unitary(circ);
+    REQUIRE((u - u_correct).cwiseAbs().sum() < ERR_EPS);
+  }
+  GIVEN("A CRz(pi) gate") {
+    Circuit circ(2);
+    Vertex v = circ.add_op<unsigned>(OpType::CRz, 1., {0, 1});
+    const Op_ptr op = circ.get_Op_ptr_from_Vertex(v);
+    Circuit rep;
+    WHEN("Default circuit replacement") { rep = CX_circ_from_multiq(op); }
+
+    REQUIRE(rep.count_gates(OpType::CX) == 1);
+
+    const auto u = tket_sim::get_unitary(rep);
+    const auto u_correct = tket_sim::get_unitary(circ);
+    REQUIRE((u - u_correct).cwiseAbs().sum() < ERR_EPS);
```

Comment:
Is this test different from "`GIVEN("A CRz(+-pi) gate") `"?

Reply:
You're right, sorry for the duplicate.